ringo_printtemplate/odfconv.py

```diff
@@ -67,6 +68,11 @@ def convert(self, data, format="ods", update=True):
         :returns: Converted data.
 
         """
+        # check document type before writing or opening
+        mimetype = check_mime_type_from_buffer(data)
+        if mimetype != "application/vnd.oasis.opendocument.text":
+            # TODO; raise error here and return. Data is not an odt file
+            return None
```

Comment:
Why do you check the file here? IMO it is sufficient to check the mimetype of the file only once inthe validator before it gets written into the DB. Once the file has passed the validator we can rely onthe correct mime-type.